src/Http/UriFactory.php

```diff
@@ -171,7 +171,7 @@ protected static function getBase(UriInterface $uri, array $server): array
 
         $docRoot = $server['DOCUMENT_ROOT'] ?? null;
```

Comment:
I am wondering if an exception should be thrown if `DOCUMENT_ROOT` is not set. Is there a valid use case where it wouldn't be?

Reply:
When I run the test suite, I get an empty string for it by the way:
```
########## DEBUG ##########
''
###########################
```
So it should never be null for sure.

Reply:
Yes, I check too and even on CLI `$_SERVER['DOCUMENT_ROOT']` is an empty string not `null`.

Reply:
Thinking about this again, defaulting `DOCUMENT_ROOT` to empty string is more user friendly as that's what you get on CLI too, so there's no need to force the user to set it.

I have made and alternate PR #18242 with test.

Reply:
It also makes upgrading from 4.x easier as per feedback provided by ishanvyas22.